Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -406,18 +405,50 @@ function New-LDAPSIdentitySource {
     else {
         Write-Host "No existing external identity sources found."
     }
-
     $Password = $Credential.GetNetworkCredential().Password
-    $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
-    [System.Array]$Certificates = 
-        foreach($CertFile in $DestinationFileArray) {
+    $DestinationFileArray = @()
+    if ($PSBoundParameters.ContainsKey('SSLCertificatesSasUrl')) {
+        $DestinationFileArray = Get-Certificates -SSLCertificatesSasUrl $SSLCertificatesSasUrl -ErrorAction Stop
+    } else {
+        $exportFolder = "$home/"
+        $remoteComputers = ,$PrimaryUrl
+        if ($PSBoundParameters.ContainsKey('SecondaryUrl')) { $remoteComputers += $SecondaryUrl }
+        
+        foreach ($computerUrl in $remoteComputers) {
+            if ($computerUrl.IndexOf('ldaps://') -ne 0) { Write-Error "Incorrect Url format entered: " + $computerUrl -ErrorAction Stop}
```

Comment:
I'd add a comment here that states you're attempting to download the cert yourself because the customer didn't pass it in. Personal preference for debugging.

Reply:
Additionally, let's add other diagnostics in this PR (at _least_ the netcat one). Later we'll add more but I think those two can go in this PR together.

The command `nc -vz {computerUrl} {portSuppliedByCustomer}` should return something like "open"